GeorgiaTech/Server/Controllers/VolumeController.cs

```diff
@@ -0,0 +1,194 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Server.Models;
+using Microsoft.EntityFrameworkCore;
+
+namespace Server.Controllers
+{
+    public class VolumeController: IVolumeController
+    {
+        private IMaterialController materialController;
+        private IAddressController addressController;
+        private readonly GTLContext _context;
+
+        public VolumeController(GTLContext context)
+        {
+            materialController = ControllerFactory.CreateMaterialController(context);
+            addressController = ControllerFactory.CreateAddressController(context);
+            _context = context;
+
+        }
+
+        /// <summary>
+        /// Create a volume entity object
+        /// </summary>
+        /// <param name="materialId"></param>
+        /// <param name="homeLocationId"></param>
+        /// <param name="currentLocationId"></param>
+        /// <returns>The created volume</returns>
+        public Volume Create(int materialId, int homeLocationId, int currentLocationId)
+        {
+     
+            try
+            {
+                var material = materialController.FindByID(materialId);
+                var homeLocation = addressController.FindByID(homeLocationId);
+                var currentLocation = addressController.FindByID(currentLocationId);
+
+                var newVolume = new Volume { Material = material, CurrentLocation = currentLocation, HomeLocation = homeLocation };
+                return newVolume;
+            }
+            catch (Exception ex)
+            {
+                throw ex;
+
+            }
+        
+        }
+        /// <summary>
+        /// Deletes a volume entry from the database
+        /// </summary>
+        /// <param name="volume"></param>
+        /// <returns>The number of rows changed</returns>
+        public int Delete(Volume volume)
+        {
+            using var transaction = _context.Database.BeginTransaction();
+
+            try
+            {
+                _context.Remove(volume);
+                var changedRows = _context.SaveChanges();
+                transaction.Commit();
```

Comment:
Savechanges already implements transactions - which means this is largely unnecessary. Simultaneously, Transactions can't be tested in an InMemoryDatabase